### ISSUE TITLE
Support for branch protection pattern

### ISSUE THREAD

**Issue by zlepper** [google/go-github#1362]
Currently it is possible to protect a set of branches using the GitHub UI by providing a pattern, rather than just a strait up branch name. However this doesn't seem to be supported in the V3, and instead seem to require the v4 api. 

Currently that is preventing https://github.com/terraform-providers/terraform-provider-github/issues/167 from being implemented.

**Comment by gmlewis** [comment]
Hi @zlepper - as far as I can tell from this page of the GitHub developer API v3 docs:
https://developer.github.com/v3/repos/branches/#update-branch-protection

It looks like only a branch name is supported and not a branch pattern, but I could be reading it wrong.

If you could please contact the excellent GitHub Tech Support team at support@github.com and ask this same question, then report back here their reply, that would be greatly appreciated.

The bottom line is that if the v3 API doesn't support this feature, then this client library won't be able to support it either, as this repo only supports the v3 API.

However, if you wish to use the v4 API, there is a separate project for that:
https://github.com/shurcooL/githubv4

Thanks!

**Comment by zlepper** [comment]
Hi @gmlewis,
I have send a support request to GitHub. I will update this ticket when they reply :)

Thank you!

**Comment by zlepper** [comment]
Reply from GitHub Developer support:

> Hey there Zlepper,
Thanks for writing in and I hope you are having a great day where you are.
Unfortunately creating / updating the branch protection rules via the REST (v3) API require you to specify exact branch matching, as oppose to being able to provide the FN-Match syntax.
As you've noted, you can use the GraphQL API with this, and we can pass this on as a feature request to the engineering team to take a look at updating this.
Sorry I can't help further here, please let us know if you have any further questions.
Thanks

In other words, i believe i would have to convince the Github Terraform people to use the other library for now. Thank you!

**Comment by gmlewis** [comment]
Wow, that was a fast reply!  OK, thank you for reporting back, @zlepper!
Closing ticket for now.  Feel free to re-open after you convince them to support this feature in v3.
:smile:
